Training3/src/task/key-logger/src/component/modal.tsx

```diff
@@ -0,0 +1,50 @@
+import React, {useEffect, useRef, useState} from 'react';
+
+type ModalProps = {
+    isOpen: boolean,
+    handleClose: () => void,
+    onKeyPress: (text: string) => void,
+}
+
+const Modal = ({isOpen, handleClose, onKeyPress}: ModalProps) => {
+    const ref = useRef<HTMLTextAreaElement>(null)
```

Comment:
why do we need use ref in this case?

Reply:
the reason I use this because I want to ref to the text area and add an event listener (In this case I wouldn't like to use dom to add event listener)

Reply:
please follow the task description, you should add a listener to the global document object, not just textarea

Reply:
I have updated. Could you check it